src/hooks/use-arrow-key-navigation.ts

```diff
@@ -94,6 +94,10 @@ export function useArrowKeyNavigation(
   const lastFocusedItem = useRef<HTMLElement | null>(null);
 
   useEffect(() => {
+    if (!containerVisible) {
+      return () => {};
+    }
```

Comment:
This is not strictly related with the other changes on this PR, but I realized we can safely skip any action when `containerVisible` is false, making the rest of the logic easier to reason about.